src/plans/gradient_plan.jl

```diff
@@ -276,20 +269,75 @@ individual one that provides these values.
 """
 abstract type DirectionUpdateRule end
 
+"""
+    DirectionUpdateRuleFactory{M,T,A,K} <: DirectionUpdateRule
+
+A generic factory to create direction update rules.
+Motivated by the fact that in some situations it is nicer to not being forced
+to specify a manifold, this factory allows to postpone that.
+It still stores all parameters one wants to provide, even those that would
+require the manifold to be filled with a reasonable default.
+Any optional parameter or keyword argument can hence still be overwritten.
+Similarly you can also pass a msnifold to the constructor, in case you
+want to use another manifolds form of defaults compared to the one of the solver.
+
+# Fields
+
+* `M::Union{Nothing,AbstractManifold}` (optional) provide a manifold for defaults
+* `args`: arguments that are passed to the `DirectionUpdateRule`
+* `kwargs`: keyword arguments to pass to the update rule upon creation
+
+# Constructor
+
+    DirectionUpdateRuleFactory(T, args...; kwargs...)
+    DirectionUpdateRuleFactory(T, M, args...; kwargs...)
+
+## Input
+
+* `T` a subtype of the `DirectionUpdateRule` this factory is meant to construct
+* `M` (optional) a manifold used for the defaults
+
+as well as arguments and keyword arguments for the update rule.
+"""
+struct DirectionUpdateRuleFactory{T,TM<:Union{AbstractManifold,Nothing},A,K}
+    M::TM
+    args::A
+    kwargs::K
+end
+function DirectionUpdateRuleFactory(
+    T::Type{<:DirectionUpdateRule}, M::TM, args...; kwargs...
+) where {TM<:AbstractManifold}
+    return DirectionUpdateRuleFactory{T,TM,typeof(args),typeof(kwargs)}(M, args, kwargs)
+end
+function DirectionUpdateRuleFactory(T::Type{<:DirectionUpdateRule}, args...; kwargs...)
+    return DirectionUpdateRuleFactory{T,Nothing,typeof(args),typeof(kwargs)}(
+        nothing, args, kwargs
+    )
+end
+function (durf::DirectionUpdateRuleFactory{T})(M::AbstractManifold) where {T}
+    return T(M, durf.args...; durf.kwargs...)
+end
+function (durf::DirectionUpdateRuleFactory{T,<:AbstractManifold})() where {T}
+    return T(durf.M, durf.args...; durf.kwargs...)
+end
+# allow both in arguments, resolve by eventually poducing a rule
+_produce_rule(::AbstractManifold, d::DirectionUpdateRule) = d
+_produce_rule(M::AbstractManifold, d::DirectionUpdateRuleFactory) = d(M)
+
 """
     IdentityUpdateRule <: DirectionUpdateRule
 
 The default gradient direction update is the identity, usually it just evaluates the gradient.
+
+Use `Gradient()` to create the corresponding factory.
 """
 struct IdentityUpdateRule <: DirectionUpdateRule end
+IdentityUpdateRule(M) = IdentityUpdateRule() # Temp hack for the mockup
```

Comment:
This looks quite fishy, maybe its required, but maybe the comment should explain a bit more why its needed or what is needed to remove the hack

Reply:
this is also one of the things I do not like too much with the factory: The factory for now _always_ “pushes” the manifold towards the `DirectionUpdateRule`

I did not yet have a goo idea what to do here. Maybe one could add a parameter to the factory, whether “on production” it hast to pass the manifold or not.
the other idea is of course to say: Any update rule _always_ has to have a constructor with at least `M::AbstractManifold` as first argument as per agreement, then this line would just have to be taped on M.

I will think about that which I like better (factory parameter or mandatory arguments in rule constructors).

Reply:
I fixed this now when reworking the factory to be more general (also working for other elements in Manopt later) and added a `requires_manifold` argument, so this line will be removed :)